Website/Providers/DataProviders/SqlDataProvider/07.04.02.SqlDataProvider

```diff
@@ -521,6 +521,174 @@ AS
 	WHERE f.WorkflowID IS NULL
 GO
 
+/* Add New OutputCache Table */
+/*****************************/
+
+IF  NOT EXISTS (SELECT * FROM dbo.sysobjects WHERE id = OBJECT_ID(N'{databaseOwner}{objectQualifier}OutputCache') AND OBJECTPROPERTY(id, N'IsTable') = 1)
+	BEGIN
+		CREATE TABLE {databaseOwner}{objectQualifier}OutputCache
+			(
+				[CacheKey] [varchar](36) NOT NULL,
+				[ItemId] [int] NOT NULL,
+				[Data] [ntext] NOT NULL,
+				[Expiration] [datetime] NOT NULL,
+				CONSTRAINT [PK_{objectQualifier}OutputCache] PRIMARY KEY CLUSTERED ( [CacheKey] ASC )
+			)
+	END
+GO
+
+/* Add New OutputCacheGetItem  */
+/*******************************/
+
+IF EXISTS (SELECT * FROM dbo.sysobjects WHERE id = object_id(N'{databaseOwner}[{objectQualifier}OutputCacheGetItem]') AND OBJECTPROPERTY(id, N'IsPROCEDURE') = 1)
+  DROP PROCEDURE {databaseOwner}{objectQualifier}OutputCacheGetItem
+GO
+	
+CREATE PROCEDURE {databaseOwner}{objectQualifier}OutputCacheGetItem
+	@CacheKey [varchar](36)
+AS
+    SELECT *
+    FROM {databaseOwner}{objectQualifier}OutputCache
+    WHERE CacheKey = @CacheKey
+GO
+
+/* Add New OutputCacheAddItem  */
+/*******************************/
+
+IF EXISTS (SELECT * FROM dbo.sysobjects WHERE id = object_id(N'{databaseOwner}[{objectQualifier}OutputCacheAddItem]') AND OBJECTPROPERTY(id, N'IsPROCEDURE') = 1)
+  DROP PROCEDURE {databaseOwner}{objectQualifier}OutputCacheAddItem
+GO
+
+CREATE PROCEDURE {databaseOwner}{objectQualifier}OutputCacheAddItem
+	@ItemId int,
+	@CacheKey [varchar](36),
+	@Data	ntext,
+	@Expiration	datetime
+AS
+    INSERT INTO {databaseOwner}{objectQualifier}OutputCache
+		(ItemId, 
+		CacheKey, 
+		Data, 
+		Expiration)
+    VALUES
+		(@ItemId, 
+		@CacheKey, 
+		@Data, 
+		@Expiration)
+GO
+
+/* Add New OutputCacheGetItemCount  */
+/************************************/
+
+IF EXISTS (SELECT * FROM dbo.sysobjects WHERE id = object_id(N'{databaseOwner}[{objectQualifier}OutputCacheGetItemCount]') AND OBJECTPROPERTY(id, N'IsPROCEDURE') = 1)
+  DROP PROCEDURE {databaseOwner}{objectQualifier}OutputCacheGetItemCount
+GO
+
+CREATE PROCEDURE {databaseOwner}{objectQualifier}OutputCacheGetItemCount
+	@ItemId int
+AS
+    SELECT COUNT(*)
+    FROM {databaseOwner}{objectQualifier}OutputCache
+    WHERE ItemId = @ItemId
```

Comment:
Would it been helpful to use "WHERE ItemId Like @ItemID", this would support wildcards (at the end of the key) to delete a key set.